src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1422,4 +1422,13 @@
   <data name="DoNotUseOutAttributeStringPInvokeParametersTitle" xml:space="preserve">
     <value>Do not use 'OutAttribute' on string parameters for P/Invokes</value>
   </data>
+  <data name="AvoidAssemblyLocationInSingleFileDescription" xml:space="preserve">
+    <value>Accessing Assembly location or file path is not valid when publishing as a single-file.</value>
```

Comment:
I don't like the word "valid" - it makes it sound like something completely wrong. Maybe just explain what it does:
"Property 'Assembly.Location' returns empty string when publishing the app as a single-file."

But I'm not a native speaker, so maybe your version actually sounds better...

Reply:
Actually maybe we should include the fact that the assembly has to be part of the app:
"Property `Assembly.Location` returns empty string for any assembly from application when published as single-file"

Reply:
Saying it returns empty string is better as it's then a condition the app can check against.

Reply:
In that case we'll probably need individual diagnostic rules for each API to describe its behavior.

Reply:
Updated

Reply:
I do agree that having one for each is lot of work - on the other hand it might give us opportunity to include some actionable feedback in the messages (if there's any). For example for `Assembly.Location` we may point users to `AppDomain.CurrentDomain.BaseDirectory` as a reasonable replacement for certain cases.